src/catalog/build.rs

```diff
@@ -195,6 +195,18 @@ fn main() -> anyhow::Result<()> {
     // 'as' is okay here because we're using it to define the type of the empty slice, which is
     // necessary since the method takes the slice as a generic arg.
     #[allow(clippy::as_conversions)]
+    // DO NOT change how JSON serialization works for these objects. The catalog relies on the JSON
```

Comment:
I don't want to generate extra work here, but wondering if we want to add some small amount of test coverage to help sand off the footgun here? Something like ben did in https://github.com/MaterializeInc/materialize/pull/23709. Certainly doesn't have to be this PR

Reply:
That's a great idea, I opened https://github.com/MaterializeInc/materialize/issues/24080 and will work on it this week.